File: server/router/survey/survey.go
package survey

import (
	"github.com/gin-gonic/gin"
	surveyApi "github.com/sxz799/surveyX/api/survey"
	"github.com/sxz799/surveyX/middleware"
	surveyService "github.com/sxz799/surveyX/service/survery"
)

func Survey(e *gin.Engine, service *surveyService.Service) {
	api := surveyApi.NewApi(service)

	g1 := e.Group("/api/admin/survey", middleware.JWTAuth())
	{
		g1.GET("/list", api.List)
		g1.POST("/", api.Add)
		g1.PUT("/", api.Update)
		g1.GET("/:id", api.Get)
		g1.DELETE("/:id", api.Del)
		g1.POST("/import", api.Import)
		g1.GET("/analysis/:id", api.Analysis)
	}

	g2 := e.Group("/api/guest/survey")
	{
		g2.GET("/:id", api.Get)
	}
}

File: web/src/api/admin/survey.js
import request from '../../utils/request.js'

export function list(params) {
    return request({
        url: '/api/admin/survey/list',
        method: 'get',
        params: params
    })
}

export function get(id) {
    return request({
        url: '/api/admin/survey/' + id,
        method: 'get',
    })
}

export function add(data) {
    return request({
        url: '/api/admin/survey',
        method: 'post',
        data
    })
}

export function update(data) {
    return request({
        url: '/api/admin/survey',
        method: 'put',
        data
    })
}

export function del(id) {
    return request({
        url: '/api/admin/survey/' + id,
        method: 'delete',
    })
}

export function analysis(id) {
    return request({
        url: '/api/admin/survey/analysis/' + id,
        method: 'get',
    })
}

export function upload(formData) {
    console.log(formData.get("file"))
    return request({
        url: '/api/admin/survey/import',
        method: 'post',
        data:formData,
    })
}